### ISSUE TITLE
Mis-calculation on 加重対空値

### ISSUE THREAD

**Issue by JohannLee** [KC3Kai/KC3Kai#2772]
### Versions:
Google Chrome | 72.0.3626.119 (Official Build) (64-bit) (cohort: Stable)
KC3 version:  33.8.3

### Actual behaviour:
Actually it is not a actual bug but some mis-calculations on 加重対空値
I was leveling at 5-2 using 噴進砲改二, but was damaged when it showed the skill rate is over 100%
So I found there is a problem in the calculations.

### Expected behaviour:
In the Screenshot below I equipped 3x +10 guns and a 噴進砲改二 which is +8, according to the formula 加重対空値 should be ((10 x 3 + 8 ) x 6 + 43) / 2 x 2 = 270. However it showed 300 actually. Thus the problem would be calculated potential anti-air of the ship. If like that it is ((10 x 3 + 8 ) x 6 + 43 + 29) / 2 x 2 = 300, which is the showed number.

### Screenshots (if applicable):
![image](https://user-images.githubusercontent.com/9301693/53691662-9d60f080-3dbd-11e9-8cb2-c8fc3e96698a.png)

**Comment by sinsinpub** [comment]
it's not bug nor miscalc. it was intentional to use the ship AA value of level 99 to compute Adjusted-AA (`加重対空値`). so yea, the value currently displayed is inaccurate for ship lower than level 99.
see: https://github.com/KC3Kai/KC3Kai/blob/master/src/library/modules/AntiAir.js#L332

because i could not remember the purpose why old guy used the lv99 value before, i have left the comments there, and kept it untouched.
it's okay for me to use either that value or ship's current AA value, since we might be rarely care about low level ship's AA ability. and i'm surprised nobody notice/report this for such a long time... 😹 
let me know if you guys really need to change it to latter one. i wanna hear more guys' opinion.

**Comment by JohannLee** [comment]
Thank you for replying so quickly.
Actually because now there is a way of leveling carriers and carrier related ships (such as BBV, AV, CAV) relying on building over 100% skill rate of 噴進砲改二, there would be some occasions (like leveling new carriers or farming 五航戦 from level 1 to level 30 for 噴進砲) to leveling ships not with max AA value. Thus I think it is still useful if it is calculated via current AA value.
Though I know the actual formula and I could still calculate by myself but I would be very appreciated if efforts could be put into this issue.

**Comment by JohannLee** [comment]
As for intentionally using max AA value I also could not come up with an explanation for it. Probably because it has been indeed no use for calculate low-level ships' AA value and in most of the time ships are with max AA value. Maybe it is for reducing unnecessary data reading? Just a guess.

**Comment by sinsinpub** [comment]
`over 100% skill rate of 噴進砲改二` seems be impossible in-game. although the formula we know for now can get a trigger rate >100%, but there are some samples reported that ship (>lv99) can be still hit by airstrike. sample amount is too low anyway...
🤔 i guess there might be a 'upper capped rate value' in game that limits it not be able to 100% proceeded.

**Comment by JohannLee** [comment]
I've been doing this for quite a period of time, and as long as the calculation is correct I never hit by airstrike. But unfortunately because it requires very frequent 出撃 so I added 5-2 into non-recording list so I cannot provide you how many 5-2's I have entered...
I've been using this method to leveling Commander Teste ,日進 and 神威(on AV form and by adding an extra slot to set enough guns), and also had farmed 五航戦 for several weeks. Until once I was leveling a 翔鶴 according to KC3's skill rate but I was hit by the airstrike.
I wonder if the problem is caused by build up just to 282. Because when I am using this method I would always build over 282 but never exactly.
Anyway I'll report if I encounter situations when building up AdjustedAA+Luck over 282 and still hit by airstrike, but as for now there is no exceptions observed in my case.
